### DISCUSSION TITLE
Copilot can't load `workerProxy.js` and crash the Extension Host

### DISCUSSION THREAD

**Discussion by JalinWang** [community/community#63737] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

The Copilot extension can't load `workerProxy.js` and crash the Extension Host(Remote) several time.

![image](https://github.com/community/community/assets/20621784/b9b438be-6692-4bf6-84ca-7ccf8f1f1a5f)

```
2023-08-15 03:37:14.863 [error] Error: Cannot find module '/home/hit/.vscode-server/extensions/github.copilot-1.102.334/dist/workerProxy.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at MessagePort.<anonymous> (node:internal/main/worker_thread:196:24)
    at MessagePort.[nodejs.internal.kHybridDispatch] (node:internal/event_target:694:20)
    at MessagePort.exports.emitMessage (node:internal/per_context/messageport:23:28)
```

**Comment by JalinWang** [comment]
After updating to the latest version `v1.102.335`, the bug still exists.

![image](https://github.com/community/community/assets/20621784/aec2b1f8-c66a-45a3-a8ae-5b81bac14207)


Corresponding dir:
![image](https://github.com/community/community/assets/20621784/9e1e7baa-a478-449f-94f0-6cba51cb85f5)

**Comment by mm-jpoole** [comment]
**TL;DR**- I was able to prevent this error by disabling `Copilot` and `Prettier` extensions, reloading vscode, then re-enabling `Copilot`.

I am having the same error. I found the logs on the remote machine (in `~/.vscode-server`) and I noticed there was a weird error from `Prettier` immediately before the error from `Copilot`.

After disabling `Prettier` and re-enabling `Copilot`, I am no longer experiencing this error. Very strange, but I will take it.

*Relevant log excerpt:*
```
[17:55:09] [127.0.0.1][ec775709][ExtensionHostConnection] <10752> Launched Extension Host Process.
[17:55:21] [127.0.0.1][ec775709][ExtensionHostConnection] <10752><stderr> 
/home/{redacted}/node_modules/prettier/plugins/flow.js:4
`),x.js_error)throw x.js_error}}else throw x}function nZ(){var x=p0.process;x&&x.on?x.on("uncaughtException",function(r,e){DK(r),x.exit(2)}):p0.addEventListener&&p0.addEventListener("error",function(r){r.error&&DK(r.error)})}nZ();function a(x,r){return(x.l>=0?x.l:x.l=x.length)==1?x(r):Ds(x,[r])}function p(x,r,e){return(x.l>=0?x.l:x.l=x.length)==2?x(r,e):Ds(x,[r,e])}function G0(x,r,e,t){return(x.l>=0?x.l:x.l=x.length)==3?x(r,e,t):Ds(x,[r,e,t])}function oe(x,r,e,t,u){return(x.l>=0?x.l:x.l=x.length)==4?x(r,e,t,u):Ds(x,[r,e,t,u])}function Je(x,r,e,t,u,i){return(x.l>=0?x.l:x.l=x.length)==5?x(r,e,t,u,i):Ds(x,[r,e,t,u,i])}function uZ(x,r,e,t,u,i,f,o){return(x.l>=0?x.l:x.l=x.length)==7?x(r,e,t,u,i,f,o):Ds(x,[r,e,t,u,i,f,o])}var QP=[Fr,jz,-1],jK=[Fr,az,-2],mn=[Fr,Q_,-3],FP=[Fr,rz,-4],w2=[Fr,dD,-7],zK=[Fr,Cj,-8],KK=[Fr,aj,-9],wr=[Fr,YU,-11],Cl=[Fr,Dj,-12],iZ=[4,0,0,0,[12,45,[4,0,0,0,0]]],HP=[0,[11,'File "',[2,0,[11,'", line ',[4,0,0,0,[11,ij,[4,0,0,0,[12,45,[4,0,0,0,[11,": ",[2,0,0]]]]]]]]]],'File "%s", line %d, characters %d-%d: %s'],Ja=[0,0,[0,0,0],[0,0,0]],Il=[0,0,0,0,1,0,0,0],JK=[0,"first_leading","last_trailing"],YK=[0,hf,gf,ni,l7,Ef,Ci,yu,uc,O7,zi,Nf,un,Ku,sc,Le,A7,F7,rf,Wf,T7,ci,Bc,hc,Gn,uf,j7,ku,ji,Fi,Jc,W7,Tu,yi,bi,_i,fi,Qf,Bi,nu,s7,Wc,K7,Xu,mc,vi,tf,Xf,Xn,ic,oc,Qi,Pe,Re,Mu,lf,dc,Mi,y7,If,ju,Gf,sf,ec,Tf,_c,Ei,ce,Ri,qn,ff,Hi,_u,e7,Cu,Li,xf,kc,cc,gu,Uu,ef,p7,$n,_f,v7,zf,z7,F1,pi,Zu,Zn,J7,o7,Pf,df,V7,xu,Pi,Ru,wu,iu,Vu,xc,Q7,jn,nc,d7,Yn,R7,Fu,Qu,Cf,Gu,af,zn,ki,jc,Yi,lu,mi,ou,Su,Ct,k7,t7,_7,R2,di,eu,Du,c7,of,cu,jf,E7,Q2,Hu,Fn,Mf,Di,uu,Xc,G7,au,Q1,Ui,Ju,Sc,tc,x7,yf,du,xi,Mc,ui,P7,n7,q7,Vn,Zf,Vf,$i,qf,L7,Ou,Wn,Ni,I7,Kn,tu,qi,ur,u7,ii,Pu,yc,$u,i7,Pc,Nc,zu,Ec,Y7,mf,g7,Ii,Of,Yf,ru,Xi,Dc,nf,Oc,r7,ti,kf,ei,bf,si,H7,Ff,bn,Yu,Sf,pn,Ic,Eu,D7,vc,wc,S7,B7,Lu,f7,a7,Oi,Nu,fc,$f,b7,Df,zc,mu,Uf,ac,su,je,pc,fu,Iu,pu,Zi,C7,Me,lc,Bu,N7,Tc,Ji,Wi,m7,Gi,Ke,Au,Z7,Rf,pf,gc,Si,wf,ze,Kc,cf,Ki,hu,rc,Bn,Qn,hi,Kf,li,Rc,h7,Lc,ai,Jn,M7,Ti,Bf,Jf,$7,w7,U7,oi,Wu,Ac,Hn,X7,qu,Yc,Ai,L2,Lf,vf,ri,U2],kn=[0,0,0];Dt(11,Cl,Dj),Dt(10,wr,YU),Dt(9,[Fr,TD,IM],TD),Dt(8,KK,aj),Dt(7,zK,Cj),Dt(6,w2,dD),Dt(5,[Fr,qM,-6],qM),Dt(4,[Fr,tM,-5],tM),Dt(3,FP,rz),Dt(2,mn,Q_),Dt(1,jK,az),Dt(0,QP,jz);var fZ="output_substring",cZ=Hf,sZ=Ts,oZ=Es,vZ="CamlinternalLazy.Undefined",aZ=mz,lZ="\\'",pZ="\\b",yZ="\\t",dZ="\\n",mZ="\\r",kZ="List.iter2",hZ="tl",_Z="hd",wZ="String.blit / Bytes.blit_string",TZ="Bytes.blit",SZ="String.sub / Bytes.sub",AZ="Array.blit",EZ="Array.sub",gZ="Map.remove_min_elt",NZ=[0,0,0,0],CZ=[0,"map.ml",400,10],IZ=[0,0,0],PZ=z4,RZ=z4,OZ=z4,LZ=z4,UZ="Stdlib.Queue.Empty",MZ="Buffer.add_substring/add_subbytes",DZ="Buffer.add: cannot grow buffer",jZ=[0,GU,93,2],zZ=[0,GU,94,2],KZ="Buffer.sub",JZ="%c",YZ="%s",WZ=DU,BZ=dz,XZ=vD,VZ=Uz,ZZ="%f",$Z="%B",QZ="%{",FZ="%}",HZ="%(",qZ="%)",GZ=PD,bZ="%t",x$="%?",r$="%r",e$="%_r",t$=[0,i2,850,23],n$=[0,i2,814,21],u$=[0,i2,815,21],i$=[0,i2,818,21],f$=[0,i2,819,21],c$=[0,i2,822,19],s$=[0,i2,823,19],o$=[0,i2,826,22],v$=[0,i2,827,22],a$=[0,i2,831,30],l$=[0,i2,832,30],p$=[0,i2,836,26],y$=[0,i2,837,26],d$=[0,i2,846,28],m$=[0,i2,847,28],k$=[0,i2,851,23],h$=[0,i2,1558,4],_$="Printf: bad conversion %[",w$=[0,i2,1626,39],T$=[0,i2,1649,31],S$=[0,i2,1650,31],A$="Printf: bad conversion %_",E$=Oj,g$=aD,N$=Oj,C$=aD,I$=[0,[11,"invalid box description ",[3,0,0]],"invalid box description %S"],P$=[0,0,4],R$=Uk,O$="neg_infinity",L$=_D,U$=Hf,M$=[0,tt],D$="%+nd",j$="% nd",z$="%+ni",K$="% ni",J$="%nx",Y$="%#nx",W$="%nX",B$="%#nX",X$="%no",V$="%#no",Z$="%nd",$$=vD,Q$="%nu",F$="%+ld",H$="% ld",q$="%+li",G$="% li",b$="%lx",xQ="%#lx",rQ="%lX",eQ="%#lX",tQ="%lo",nQ="%#lo",uQ="%ld",iQ=dz,fQ="%lu",cQ="%+Ld",sQ="% Ld",oQ="%+Li",vQ="% Li",aQ="%Lx",lQ="%#Lx",pQ="%LX",yQ="%#LX",dQ="%Lo",mQ="%#Lo",kQ="%Ld",hQ=Uz,_Q="%Lu",wQ="%+d",TQ="% d",SQ="%+i",AQ="% i",EQ="%x",gQ="%#x",NQ="%X",CQ="%#X",IQ="%o",PQ="%#o",RQ=hl,OQ=DU,LQ=qD,UQ=oa,MQ="@}",DQ="@?",jQ=`@
                                     ^

[17:55:21] [127.0.0.1][ec775709][ExtensionHostConnection] <10752><stderr> Error: Cannot find module '/home/{redacted}/.vscode-server/extensions/github.copilot-1.102.336/dist/workerProxy.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at MessagePort.<anonymous> (node:internal/main/worker_thread:196:24)
    at MessagePort.[nodejs.internal.kHybridDispatch] (node:internal/event_target:694:20)
    at MessagePort.exports.emitMessage (node:internal/per_context/messageport:23:28)
Emitted 'error' event on Worker instance at:
    at Worker.[kOnErrorMessage] (node:internal/worker:290:10)
    at Worker.[kOnMessage] (node:internal/worker:301:37)
    at MessagePort.<anonymous> (node:internal/worker:202:57)
    at MessagePort.[nodejs.internal.kHybridDispatch] (node:internal/event_target:694:20)
    at MessagePort.exports.emitMessage (node:internal/per_context/messageport:23:28) {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

[17:55:21] [127.0.0.1][ec775709][ExtensionHostConnection] <10752> Extension Host Process exited with code: 7, signal: null.
Cancelling previous shutdown timeout
[17:55:21] Cancelling previous shutdown timeout
Last EH closed, waiting before shutting down
[17:55:21] Last EH closed, waiting before shutting down
[17:55:21] [127.0.0.1][ec775709][ExtensionHostConnection] Unknown reconnection token (seen before).
```

**Comment by gonzalologorg** [comment]
I've removed prettier but still occurs in my IDE
To fix it, i rolled back to 1.101.321, anything after that, will request workerProxy.js (Which doesn't exists even on old versions)

**Comment by hsenag** [comment, marked as answer]
Hi, sorry about these errors. We hope they're fixed in 1.103.
